app/controllers/cinema_halls_controller.rb

```diff
@@ -5,40 +5,37 @@ class CinemaHallsController < ApplicationController
 
   def index
     authorize CinemaHall
-    cinema_halls = CinemaHall.all
-
-    render jsonapi: cinema_halls, except: blacklisted_attributes
+    render jsonapi: CinemaHalls::UseCases::FetchAll.new.call, except: blacklisted_attributes
   end
 
   def show
-    cinema_hall = CinemaHall.find(params[:id])
+    cinema_hall = CinemaHalls::UseCases::FetchOne.new.call(id: params[:id])
     authorize cinema_hall
-
     render jsonapi: cinema_hall, except: blacklisted_attributes
   end
 
   def create
-    cinema_hall = CinemaHall.new(cinema_hall_params)
+    cinema_hall = CinemaHalls::UseCases::Create.new.call(params: cinema_hall_params)
     authorize cinema_hall
-    if cinema_hall.save
+    if cinema_hall.valid?
       render jsonapi: cinema_hall, status: :created
     else
       render jsonapi: cinema_hall.errors, status: :unprocessable_entity
     end
   end
 
   def update
-    cinema_hall = CinemaHall.find(params[:id])
+    cinema_hall = CinemaHalls::UseCases::Update.new.call(id: params[:id], params: cinema_hall_params)
     authorize cinema_hall
-    if cinema_hall.update(cinema_hall_params)
+    if cinema_hall.valid?
       render jsonapi: cinema_hall
     else
       render jsonapi: cinema_hall.errors, status: :unprocessable_entity
     end
   end
 
   def destroy
-    cinema_hall = CinemaHall.find(params[:id])
+    cinema_hall = CinemaHalls::UseCases::Delete.new.call(id: params[:id])
```

Comment:
Here is the same problem - even if you don't have access to object, you can remove it.